### ISSUE TITLE
[bug] `@eslint-react/no-unstable-context-value` false positive for `React.FunctionComponent` and value in props

### ISSUE THREAD

**Issue by kachkaev** [Rel1cx/eslint-react#966]
### Describe the bug

I found a strange edge case after upgrading to [v1.30.0](https://github.com/Rel1cx/eslint-react/releases/tag/v1.30.0). `naming-convention/context-name` was off for me until then because it was not in the `recommended` preset.

### Reproduction

```ts
const MyContext = React.createContext<string>("");

export function MyFunctionComponent({ children, x }: { children: React.ReactNode; x: string }) {
  return <MyContext value={x}>{children}</MyContext>;
}

export const MyConstComponent: React.FunctionComponent<{
  children: React.ReactNode;
  x: string;
}> = ({ children, x }) => {
  return <MyContext value={x}>{children}</MyContext>;
};
```

### Expected behavior

no errors (context value is `string` in both cases)

**Observed**

```
A/an 'arrow function expression' passed as the value prop to the context provider should not be constructed. It will change on every render. Consider wrapping it in a useCallback hook.eslint[@eslint-react/no-unstable-context-value](https://eslint-react.xyz/docs/rules/no-unstable-context-value)
```

Error location:

```
({ children, x }) => {
  return <MyContext value={x}>{children}</MyContext>;
}
```

Interestingly, it only occurs for `export const MyConstComponent` and not for `export function MyFunctionComponent`

### Platform and versions

```Text
"@eslint-react/eslint-plugin": "1.29.0",
"eslint": "9.21.0",
"typescript-eslint": "8.26.0"
"typescript": "5.8.2"
```

### Stack trace

```Text

```

**Comment by kachkaev** [comment]
Thank you @Rel1cx! `1.30.1-beta.0` worked like a charm! 👏
